frontend/src/pages/modelServing/screens/projects/kServeModal/EnvironmentVariablesSection.tsx

```diff
@@ -16,31 +16,44 @@ import {
 } from '@patternfly/react-icons';
 import { UpdateObjectAtPropAndValue } from '~/pages/projects/types';
 import { CreatingInferenceServiceObject } from '~/pages/modelServing/screens/types';
+import { ServingContainer } from '~/k8sTypes';
 
 type EnvironmentVariablesSectionType = {
-  data?: CreatingInferenceServiceObject;
+  data: CreatingInferenceServiceObject;
   setData: UpdateObjectAtPropAndValue<CreatingInferenceServiceObject>;
 };
 
-const EnvironmentVariablesSection: React.FC<EnvironmentVariablesSectionType> = () => {
-  const [additionalEnvVars, setAdditionalEnvVars] = React.useState<
-    Array<{ name: string; value: string }>
-  >([]);
+const EnvironmentVariablesSection: React.FC<EnvironmentVariablesSectionType> = ({
+  data,
+  setData,
+}) => {
+  const [additionalEnvVars, setAdditionalEnvVars] = React.useState<ServingContainer['env']>(
+    data.servingRuntimeEnvVars,
+  );
 
   const addEnvVar = () => {
-    setAdditionalEnvVars((prevVars) => [...prevVars, { name: '', value: '' }]);
+    if (additionalEnvVars) {
```

Comment:
This condition is making it so that if `additionalEnvVars` is undefined (which ends up happening if we are editing a model that was deployed before this PR), clicking the "Add variable" button doesn't do anything. I think we should remove this `if` statement and use `(additionalEnvVars || [])` here.

Now, for the other two cases (`removeEnvVar` and `updateEnvVar`) we know there will be at least one env var in state if we can reach those functions, so it's probably fine to leave the `if` statement for those, especially since the update function expects `additionalEnvVars[indexToUpdate]` to be defined.